testkit/src/lib.rs

```diff
@@ -634,29 +624,52 @@ impl TestKit {
     ///
     /// # Panics
     ///
-    /// - Panics in the case any of transaction hashes are not in the mempool.
+    /// - Panics in the case any of transaction hashes are not in the pool.
     pub fn create_block_with_tx_hashes(&mut self, tx_hashes: &[crypto::Hash]) {
         self.poll_events();
 
         {
-            let txs = self.mempool();
+            let snapshot = self.blockchain.snapshot();
+            let schema = CoreSchema::new(&snapshot);
             for hash in tx_hashes {
-                assert!(txs.contains_key(hash));
+                assert!(schema.transactions_pool().contains(hash));
             }
         }
 
         self.do_create_block(tx_hashes);
     }
 
-    /// Creates block with all transactions in the mempool.
+    /// Creates block with all transactions in the pool.
     pub fn create_block(&mut self) {
         self.poll_events();
 
-        let tx_hashes: Vec<_> = self.mempool().keys().cloned().collect();
-
+        let snapshot = self.blockchain.snapshot();
+        let schema = CoreSchema::new(&snapshot);
+        let txs = schema.transactions_pool();
+        let tx_hashes: Vec<_> = txs.iter().collect();
+        //FIXME: every block should contain two merges
+        {
+            let blockchain = self.blockchain_mut();
+            let fork = blockchain.fork();
+            blockchain.merge(fork.into_patch()).unwrap();
+        }
         self.do_create_block(&tx_hashes);
     }
 
+    /// Adds transaction into persistent pool.
+    pub fn add_tx(&mut self, transaction: RawMessage) {
+        let mut fork = self.blockchain.fork();
+        let mut schema = CoreSchema::new(&mut fork);
+        schema.add_transaction_into_pool(transaction)
+    }
+
+    /// Checks if transaction can be found in pool
+    pub fn is_tx_in_pool(&self, tx_hash: &Hash) -> bool {
```

Comment:
I'd rather return the (wrapper around?) the readonly pool index, so the user can do more operations with it if he chooses to.